src/infer/browsers/inferSafariVersion.ts

```diff
@@ -0,0 +1,72 @@
+import axios from 'axios';
+import * as log from 'loglevel';
+
+export type SafariVersion = {
+  majorVersion: number;
+  version: string;
+  webkitVersion: string;
+};
+
+const SAFARI_VERSIONS_HISTORY_URL =
+  'https://en.wikipedia.org/wiki/Safari_version_history';
+
+export async function getLatestSafariVersion(
+  url = SAFARI_VERSIONS_HISTORY_URL,
+): Promise<SafariVersion> {
+  log.debug('Grabbing apple version data from', url);
+  const response = await axios.get(url, { timeout: 5000 });
+  if (response.status !== 200) {
+    throw new Error(`Bad request: Status code ${response.status}`);
+  }
+
+  try {
+    // This would be easier with an HTML parser, but we're not going to include an extra dependency for something that dumb
+    const rawData: string = response.data;
+
+    const majorVersions = [
+      ...rawData.matchAll(
+        /class="mw-headline" id="Safari_[0-9]*">Safari ([0-9]*)</g,
+      ),
+    ].map((match) => match[1]);
+
+    const majorVersion = parseInt(majorVersions[majorVersions.length - 1]);
+
+    const majorVersionTable = rawData
+      .split('>Release history<')[2]
+      .split('<table')
+      .filter((table) => table.includes(`Safari ${majorVersion}.x`))[0];
+
+    const versionRows = majorVersionTable.split('<tbody')[1].split('<tr');
+
+    let version = null;
+    let webkitVersion: string = null;
+
+    for (const versionRow of versionRows.reverse()) {
+      const versionMatch = [
+        ...versionRow.matchAll(/>\s*(([0-9]*\.){2}[0-9])\s*</g),
+      ];
+      if (versionMatch.length > 0 && !version) {
+        version = versionMatch[0][1];
+      }
+
+      const webkitVersionMatch = [
+        ...versionRow.matchAll(/>\s*(([0-9]*\.){3,4}[0-9])\s*</g),
+      ];
+      if (webkitVersionMatch.length > 0 && !webkitVersion) {
+        webkitVersion = webkitVersionMatch[0][1];
+      }
+      if (version && webkitVersion) {
+        break;
+      }
+    }
+
```

Comment:
Eeek :grimacing: .

What about simply having a `DEFAULT_SAFARI_VERSION` in const.ts, with a link to the Wikipedia page, and a comment reminding to bump it when we bump Electron?

Alternatively, if you insist on keeping this code, we can still have the fallback with `DEFAULT_SAFARI_VERSION` frmo const.ts in the `catch` below (as I suggested for Firefox).

Reply:
> Eeek 😬 .

Yeah. 😭 

> What about simply having a `DEFAULT_SAFARI_VERSION` in const.ts, with a link to the Wikipedia page, and a comment reminding to bump it when we bump Electron?
> 
> Alternatively, if you insist on keeping this code, we can still have the fallback with `DEFAULT_SAFARI_VERSION` frmo const.ts in the `catch` below (as I suggested for Firefox).

I insist on nothing here. 😆 

Here's what it comes down to for me, and I'll let you make the call. My only goal here is to limit the pain of keeping something up to date and generating user agents. If this code keeps us from having to keep another constant up to date, good. But if this code is so brittle (and it could be depending on the whims of a Wikipedia editor) that it breaks and then we have to support THAT, then it's not good.

So basically we just have to make a bet on which one is easier to maintain (so probably just a constant, but I'll let you make the call).

Also one thing I was trying to keep in mind here is that I don't want someone looking to block Nativefier to be able to easily guess what the user agent would be here. But that may be paranoia. 🤣

Reply:
@TheCleric

1. Okay to give it a try, we'll see how long it works...
2. ... with fallback to a constant in the `catch`
2. ... and with more logging leaving a trace of what happened, so that it's auditable by users. Something like "Trying to get Safari versions from the web... \n Failed to get Safari version from the web, falling back to $var_from_const_dot_ts"